src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -0,0 +1,86 @@
+package com.github.sdonkov;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class PrefixTreeTest {
+
+    private static List<CharSequence> words;
+    private static PrefixTree prefixTree;
+
+    @BeforeAll
+    static void setUp() throws IOException {
+        words = new ArrayList<>();
+        prefixTree = new PrefixTree();
+        try (InputStream is = PrefixTreeTest.class.getResourceAsStream("/tests.txt")) {
+            if (is != null) {
+                Scanner sc = new Scanner(is);
```

Comment:
An alternative to the `Scanner` in this case is the [readAllLines](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#readAllLines-java.nio.file.Path-) method, which would be nicer as code-readers won't have to figure out what do you need a `Scanner` for (it has other uses). Unlike the `Scanner`, it also ties nicely with lambdas, which you can utilize here too, instead of looping.